tests/framework/test_utilities.py

```diff
@@ -29,10 +31,19 @@ def test_to_yearly() -> None:
     assert round(new_rate, 5) == round(0.01, 5)
 
 
-def test_rate_to_probability() -> None:
-    rate = np.array([0.001])
-    prob = rate_to_probability(rate)
-    assert np.isclose(prob, 0.00099950016662497809)
+@pytest.mark.parametrize(
+    "rate, time_scaling_factor",
+    [
+        (0.5, 1),
+        (0.25, 1.0),
+        (0.5, 0.5),
+        (0.25, 0.5),
+    ],
+)
+def test_rate_to_probability(rate: float, time_scaling_factor: float) -> None:
```

Comment:
Need tests of calls where the `time_scaling_factor` is not provided for both directions.